agent/engine/task_manager.go

```diff
@@ -489,8 +494,27 @@ func (mtask *managedTask) cleanupTask(taskStoppedDuration time.Duration) {
 		cleanupTimeBool <- true
 		close(cleanupTimeBool)
 	}()
+	// wait for the cleanup time to elapse, signalled by cleanupTimeBool
 	for !mtask.waitEvent(cleanupTimeBool) {
 	}
+	stoppedSentBool := make(chan bool)
+	go func() {
```

Comment:
Without context, there is no way to stop this potential "long run"  NEW go routine (which could run for at worst 72 hours).

In case if there is another kind of state mismatch between agent and backend, backend thinks this instance is able to launch new task but agent is holding those "long run" cleanup GO routines.  Is it possible, that agent could run out of memory ...?